### PR TITLE
[MISC] Speed up simulation with IMU sensors.

### LINKED ISSUE #2111: [Bug]: IMU halves sim speed
### Bug Description

Adding a gs.sensors.IMU in the setup slices sim speed by a factor of 1/2. Is this expected?

I suspected that the acceleration is readily available from the solver and it wouldn't be so expensive to get access to it. I removed the noise parameters from the script as the performance drop is the same.

### Steps to Reproduce

```python
import genesis as gs
def main():

    ########################## init ##########################
    gs.init(backend=gs.gpu)

    ########################## create a scene ##########################
    scene = gs.Scene(
        sim_options=gs.options.SimOptions(
            dt=0.01,
        ),
        vis_options=gs.options.VisOptions(
            show_world_frame=False,
        ),
        viewer_options=gs.options.ViewerOptions(
            camera_pos=(3.5, 0.0, 2.5),
            camera_lookat=(0.0, 0.0, 0.5),
            camera_fov=40,
        ),
        profiling_options=gs.options.ProfilingOptions(
            show_FPS=True,
        ),
        show_viewer=False,
    )

    ########################## entities ##########################
    scene.add_entity(gs.morphs.Plane())
    franka = scene.add_entity(
        gs.morphs.MJCF(file="xml/franka_emika_panda/panda.xml"),
    )
    end_effector = franka.get_link("hand")

    ########################## record sensor data ##########################
    imu = scene.add_sensor(
        gs.sensors.IMU(
            entity_idx=franka.idx,
            link_idx_local=end_effector.idx_local,
            pos_offset=(0.0, 0.0, 0.15),
        )
    )

    ########################## build ##########################
    scene.build(n_envs=1000)

    steps = 5000
    for _ in range(steps):
        scene.step()
        data = imu.read()

if __name__ == "__main__":
    main()
```
Use the code from above, a stripped down version of the `franka_imu` example.

### Expected Behavior

I do not know if my intuition is correct here, but I did not expect the performance drop to be so severe.

### Screenshots/Videos

_No response_

### Relevant log output

```shell
With IMU:
[Genesis] [19:07:07] [INFO] Running at 330,342.83 FPS (330.34 FPS per env, 1000 envs).
[Genesis] [19:07:07] [INFO] Running at 330,811.04 FPS (330.81 FPS per env, 1000 envs).
[Genesis] [19:07:07] [INFO] Running at 330,708.96 FPS (330.71 FPS per env, 1000 envs).
[Genesis] [19:07:07] [INFO] Running at 331,146.60 FPS (331.15 FPS per env, 1000 envs).
[Genesis] [19:07:07] [INFO] Running at 331,436.14 FPS (331.44 FPS per env, 1000 envs).
[Genesis] [19:07:07] [INFO] Running at 331,535.34 FPS (331.54 FPS per env, 1000 envs).
[Genesis] [19:07:07] [INFO] Running at 331,918.55 FPS (331.92 FPS per env, 1000 envs).

Without IMU:
[Genesis] [19:05:09] [INFO] Running at 732,527.27 FPS (732.53 FPS per env, 1000 envs).
[Genesis] [19:05:09] [INFO] Running at 731,397.81 FPS (731.40 FPS per env, 1000 envs).
[Genesis] [19:05:09] [INFO] Running at 730,217.20 FPS (730.22 FPS per env, 1000 envs).
[Genesis] [19:05:09] [INFO] Running at 729,065.14 FPS (729.07 FPS per env, 1000 envs).
[Genesis] [19:05:09] [INFO] Running at 727,863.48 FPS (727.86 FPS per env, 1000 envs).
[Genesis] [19:05:09] [INFO] Running at 724,943.83 FPS (724.94 FPS per env, 1000 envs).
[Genesis] [19:05:09] [INFO] Running at 722,213.92 FPS (722.21 FPS per env, 1000 envs).
```

### Environment

OS: Ubuntu 22.04 LTS
GPU: NVIDIA A5500 RTX
CPU: 13th Gen Intel(R) Core(TM) i9-13900K
GPU-driver version: 470
CUDA: 12.8

### Release version or Commit ID

Current main branch version 3.9
ID: 7879b015ce3f113f5a2878e0da005342c2e0728b

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by duburcqa**
## Description

The IMU used to compute its raw data through four solver getters followed by a chain of tensor operations, spending more host time per step than a whole simulation step of a thousand environments. It is now a single kernel reading the link state directly. Two supporting changes come with it: the per-sensor metadata table of a lone sensor is materialized, so it can be indexed by a kernel and written one environment at a time, and the sensor ring buffer keeps its rotation index in host memory, so selecting a slot no longer blocks on the device queue.

## Related Issue

Resolves Genesis-Embodied-AI/Genesis#2111

## Motivation and Context

Adding a single IMU to a Franka scene with 1000 environments on CUDA took the step time from 0.81 ms to 2.40 ms, tripling it. It now costs 1.33 ms, i.e. 416 to 753 FPS per environment, the sensor itself accounting for 0.52 ms instead of 1.60 ms. Keeping the ring rotation index on host memory pays off for every sensor type rather than the IMU alone: measured with a sensor doing no work of its own, it cuts the shared per-step pipeline from 0.334 ms to 0.209 ms, worth 11% of the step with one sensor and 7% with sixteen. As a side effect, `set_pos_offset` and `set_quat_offset` no longer raise on a scene holding a single sensor and more than one environment.

## How Has This Been / Can This Be Tested?

`pytest tests/sensors` covers all of it. `test_lone_sensor_resolution_and_offset_setter` fails before this change as soon as there is more than one environment, and the box now settles before the assertions so that the quantization check applies to a non-zero acceleration rather than trivially to zero. Readings were compared against the previous implementation on a scene driving three IMUs mounted on different links with position and orientation offsets, agreeing to 2e-6 in single precision.

## Checklist:

- [x] I read the **CONTRIBUTING** document.
- [x] I followed the `Submitting Code Changes` section of **CONTRIBUTING** document.
- [x] I tagged the title correctly (including BUG FIX/FEATURE/MISC/BREAKING)
- [x] I updated the [documentation](https://github.com/Genesis-Embodied-AI/genesis-doc) accordingly or no change is needed.
- [x] I tested my changes and added instructions on how to test it for reviewers.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: genesis/utils/ring_buffer.py [hunk 1]
```diff
@@ -37,11 +43,10 @@ def __init__(
             self.buffer = buffer
         self.N = N
         if idx is None:
-            self._idx = torch.tensor(-1, dtype=torch.int64, device=gs.device)
+            self._idx = torch.tensor(-1, dtype=torch.int64, device="cpu")
```
[the remark this answers is not in the document]
Comment (duburcqa) [reply genesis/utils/ring_buffer.py]: A 0-dim host tensor is a valid scalar operand of a device tensor, so `(self._idx - idx) % self.N` follows the device of `idx`: verified on CUDA and Metal that both the per-row delay gather and the history gather return the correct slots on device, and `tests/sensors` passes on the gpu backend.
